src/command_modules/azure-cli-sql/tests/test_sql_commands.py

```diff
@@ -3,101 +3,128 @@
-
-    def __init__(self, test_method):
-        super(SqlServerMgmtScenarioTest, self).__init__(__file__, test_method,
-                                                        resource_group='cli-test-sql-mgmt')
-        self.sql_server_names = ['cliautomation01', 'cliautomation02']
-        self.location = "westus"
-        self.admin_login = 'admin123'
-        self.admin_passwords = ['SecretPassword123', 'SecretPassword456']
-
-    def test_sql_server_mgmt(self):
-        self.execute()
+from azure.cli.testsdk.base import execute
+from azure.cli.testsdk.exceptions import CliTestError
+from azure.cli.testsdk import (
+    JMESPathCheck,
+    NoneCheck,
+    ResourceGroupPreparer,
+    ScenarioTest)
+from azure.cli.testsdk.preparers import (
+    AbstractPreparer,
+    SingleValueReplacer)
+
+
+# Constants
+server_name_prefix = 'clitestserver'
+server_name_max_length = 63
+
+
+class SqlServerPreparer(AbstractPreparer, SingleValueReplacer):
+    # pylint: disable=too-many-arguments
+    def __init__(self, name_prefix=server_name_prefix, parameter_name='server', location='westus',
+                 admin_user='admin123', admin_password='SecretPassword123',
+                 resource_group_parameter_name='resource_group', skip_delete=True):
+        super(SqlServerPreparer, self).__init__(name_prefix, server_name_max_length)
+        self.location = location
+        self.parameter_name = parameter_name
+        self.admin_user = admin_user
+        self.admin_password = admin_password
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.skip_delete = skip_delete
+
+    def create_resource(self, name, **kwargs):
+        group = self._get_resource_group(**kwargs)
+        template = 'az sql server create -l {} -g {} -n {} -u {} -p {}'
+        execute(template.format(self.location, group, name, self.admin_user, self.admin_password))
+        return {self.parameter_name: name}
+
+    def remove_resource(self, name, **kwargs):
+        if not self.skip_delete:
+            group = self._get_resource_group(**kwargs)
+            execute('az sql server delete -g {} -n {} --yes --no-wait'.format(group, name))
+
+    def _get_resource_group(self, **kwargs):
+        try:
+            return kwargs.get(self.resource_group_parameter_name)
+        except KeyError:
+            template = 'To create a sql server account a resource group is required. Please add ' \
+                       'decorator @{} in front of this storage account preparer.'
+            raise CliTestError(template.format(ResourceGroupPreparer.__name__,
+                                               self.resource_group_parameter_name))
+
+
+class SqlServerMgmtScenarioTest(ScenarioTest):
+
+    @ResourceGroupPreparer()
+    def test_sql_server_mgmt(self, resource_group, resource_group_location):
+        servers = ['cliautomation51', 'cliautomation52']  # TODO: Server names should be randomized
```

Comment:
You can use `self.create_random_name`: https://github.com/Azure/azure-cli/blob/master/src/azure-cli-testsdk/azure/cli/testsdk/base.py#L85